apis/k8ssandra/v1alpha1/cassandraconfig_types.go

```diff
@@ -1041,3 +1049,106 @@ type TrackWarnings struct {
 	LocalReadSize       *int `json:"local_read_size,omitempty"`
 	RowIndexSize        *int `json:"row_index_size,omitempty"`
 }
+
+type ClientEncryptionOptions struct {
+	Enabled bool `json:"enabled"`
+
+	// +optional
+	Optional *bool `json:"optional,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Keystore *string `json:"keystore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	KeystorePassword *string `json:"keystore_password,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Truststore *string `json:"truststore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	TruststorePassword *string `json:"truststore_password,omitempty"`
+
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// +optional
+	AcceptedProtocols *[]string `json:"accepted_protocols,omitempty"`
+
+	// +optional
+	Algorithm *string `json:"algorithm,omitempty"`
+
+	// +optional
+	StoreType *string `json:"store_type,omitempty"`
+
+	// +optional
+	CipherSuites *[]string `json:"cipher_suites,omitempty"`
+
+	// default: false
+	// +optional
+	RequireClientAuth *bool `json:"require_client_auth,omitempty"`
```

Comment:
Typically I find that pointers are unnecessary for bool and string. We should use value types if possible. Makes the code easier to manage.

Reply:
sure thing, I've removed pointers that were not absolutely necessary.